macros/develop.sql

```diff
@@ -0,0 +1,79 @@
+{% macro develop(develop_yml, grain, dimensions=[], secondary_calculations=[], start_date=None, end_date=None, where=None) -%}
+    {{ return(adapter.dispatch('develop', 'metrics')(develop_yml, grain, dimensions, secondary_calculations, start_date, end_date, where)) }}
+{% endmacro %}
+
+
+{% macro default__develop(develop_yml, grain, dimensions=[], secondary_calculations=[], start_date=None, end_date=None, where=None) -%}
+    -- Need this here, since the actual ref is nested within loops/conditions:
+    -- depends on: {{ ref(var('dbt_metrics_calendar_model', 'dbt_metrics_default_calendar')) }}
+
+    {%- if not execute %}
+        {%- do return("not execute") %}
+    {%- endif %}
+    
+    {# ############
+    VALIDATION OF PROVIDED YML
+    ############ #}
+    {% set develop_yml = fromyaml(develop_yml) %}
+
+    {% if develop_yml["metrics"] | length > 1%}
+        {%- do exceptions.raise_compiler_error("The develop macro only supports testing a single macro.") %}
+    {% endif %}
+
+    {% set metric_definition = develop_yml["metrics"][0] %}
+
+    {%- if not metric_definition["name"] %}
+        {%- do exceptions.raise_compiler_error("The provided yml is missing a name") %}
+    {%- endif %}
+
+    {%- if not metric_definition["model"] %}
+        {%- do exceptions.raise_compiler_error("The provided yml is missing a model") %}
+    {%- endif %}
+
+    {%- if not metric_definition["timestamp"] %}
+        {%- do exceptions.raise_compiler_error("The provided yml is missing a timestamp") %}
+    {%- endif %}
+
+    {%- if not metric_definition["time_grains"] %}
+        {%- do exceptions.raise_compiler_error("The provided yml is missing time grains") %}
+    {%- endif %}
+
+    {%- if grain not in metric_definition["time_grains"] %}
+        {%- do exceptions.raise_compiler_error("The macro provided grain is missing from the metric definition yml") %}
```

Comment:
This sentence doesn't make sense to me. What is a macro provided grain?

Reply:
The macro provided grain is the grain used in the macro call. 

> from {{ metrics.develop(
        develop_yml=my_metric_yml,
        grain='month'

IE Month. But can definitely see how that isn't the cleanest way to explain that. Let me see if I can rewrite

Reply:
Updated to:

> "The selected grain is missing from the metric definition yml"